climada_petals/entity/impact_funcs/river_flood.py

```diff
@@ -55,154 +53,452 @@ def __init__(self):
-        elif region.lower().replace(" ", "") == "northamerica":
-            impf.id = 4
-            impf.name = "Flood North America JRC Residential noPAA"
-            impf.continent = 'NorthAmerica'
-            impf.intensity = np.array([0., 0.1, 0.5, 1., 1.5, 2., 3., 4., 5.,
-                                      6., 12.])
-            impf.mdd = np.array([0.0000, 0.2018, 0.4433, 0.5828, 0.6825, 0.7840,
-                                 0.8543, 0.9237, 0.9585, 1.0000, 1.0000])
-            impf.mdr = np.array([0.0000, 0.2018, 0.4433, 0.5828, 0.6825, 0.7840,
-                                 0.8543, 0.9237, 0.9585, 1.0000, 1.0000])
-        elif region.lower() == "oceania":
-            impf.id = 5
-            impf.name = "Flood Oceania JRC Residential noPAA"
-            impf.continent = 'Oceania'
-            impf.intensity =  np.array([0., 0.5, 1., 1.5, 2., 3., 4., 5., 6., 12.])
-            impf.mdd = np.array([0.00, 0.48, 0.64, 0.71, 0.79, 0.93, 0.97, 0.98,
-                                 1.00, 1.00])
-            impf.mdr = np.array([0.000, 0.480, 0.640, 0.710, 0.790, 0.930, 0.970,
-                                 0.980, 1.000, 1.000])
-        elif region.lower().replace(" ", "") == "southamerica":
-            impf.id = 6
-            impf.name = "Flood South America JRC Residential noPAA"
-            impf.continent = 'SouthAmerica'
-            impf.intensity =  np.array([0., 0.5, 1., 1.5, 2., 3., 4., 5., 6., 12.])
-            impf.mdd = np.array([0.0000, 0.4908, 0.7112, 0.8420, 0.9494,
-                                 0.9836, 1.0000, 1.0000, 1.0000, 1.0000])
-            impf.mdr = np.array([0.0000, 0.4908, 0.7112, 0.8420, 0.9494, 0.9836,
-                                 1.0000, 1.0000, 1.0000, 1.0000])
+
+        if sector == 'residential':
+            impf_values, impf_id = from_jrc_impf_residential(region)
+
+        elif sector == 'industrial':
+            impf_values, impf_id = from_jrc_impf_industrial(region)
+
+        elif sector == 'commercial':
+            impf_values, impf_id = from_jrc_impf_commercial(region)
+
+        elif sector == 'transport':
+            impf_values, impf_id = from_jrc_impf_transport(region)
+
+        elif sector == 'infrastructure':
+            impf_values, impf_id = from_jrc_impf_infrastructure(region)
+
+        elif sector == 'agriculture':
+            impf_values, impf_id = from_jrc_impf_agriculture(region)
+
         else:
-            raise ValueError(f"Unrecognized world region: {region}")
+            raise ValueError(f"Unrecognized sector: {sector}")
+
+        impf = cls()
+        impf.name = f"Flood {region} JRC {sector.capitalize()} noPAA"
+        impf.continent = f"{region}"
+        impf.id = impf_id
+        impf.mdd = impf_values
+        impf.mdr = impf_values
+        impf.intensity = np.array([0., 0.05, 0.5, 1., 1.5, 2., 3., 4., 5., 6., 12.])
         impf.paa = np.ones(len(impf.intensity))
+
         return impf
```

Comment:
Please use the init of the class properly. The init variables are:

```
        id: Union[str, int] = "",
        intensity: Optional[np.ndarray] = None,
        mdd: Optional[np.ndarray] = None,
        paa: Optional[np.ndarray] = None,
        continent: str = ''
        name: str = ""

So, the code should be 

return cls(
    id = impf_id,
    mdd = impf_values,
    paa = np.ones(len(intensity)),
    intensity = intensity,
    name = name,
    continent=continent
)
```

Reply:
Also, ImpactFunc does not have an attribute `.mdr`. Please make sure to remove this.

Reply:
removed